homeassistant/components/openuv/__init__.py

```diff
@@ -43,10 +44,79 @@
 NOTIFICATION_ID = "openuv_notification"
 NOTIFICATION_TITLE = "OpenUV Component Setup"
 
-TOPIC_UPDATE = f"{DOMAIN}_data_update"
-
 PLATFORMS = [Platform.BINARY_SENSOR, Platform.SENSOR]
 
+SERVICE_NAME_UPDATE_DATA = "update_data"
+SERVICE_NAME_UPDATE_PROTECTION_DATA = "update_protection_data"
+SERVICE_NAME_UPDATE_UV_INDEX_DATA = "update_uv_index_data"
+
+SERVICES = (
+    SERVICE_NAME_UPDATE_DATA,
+    SERVICE_NAME_UPDATE_PROTECTION_DATA,
+    SERVICE_NAME_UPDATE_UV_INDEX_DATA,
+)
+
+SERVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_DEVICE_ID): cv.string,
+    }
+)
+
+
+@callback
+def async_get_openuv_for_service_call(hass: HomeAssistant, call: ServiceCall) -> OpenUV:
+    """Get the OpenUV object related to a service call (by device ID)."""
+    device_id = call.data[CONF_DEVICE_ID]
+    device_registry = dr.async_get(hass)
+
+    if (device_entry := device_registry.async_get(device_id)) is None:
+        raise ValueError(f"Invalid OpenUV service ID: {device_id}")
+
+    for entry_id in device_entry.config_entries:
+        if (entry := hass.config_entries.async_get_entry(entry_id)) is None:
+            continue
+        if entry.domain == DOMAIN:
+            return cast(OpenUV, hass.data[DOMAIN][entry_id])
+
+    raise ValueError(f"No OpenUV object for service ID: {device_id}")
+
+
+@callback
+def async_log_deprecated_service_call(
+    hass: HomeAssistant,
+    call: ServiceCall,
+    alternate_service: str,
+    alternate_target: str,
+    breaks_in_ha_version: str,
+) -> None:
+    """Log a warning about a deprecated service call."""
+    deprecated_service = f"{call.domain}.{call.service}"
+
+    async_create_issue(
+        hass,
+        DOMAIN,
+        f"deprecated_service_{deprecated_service}",
+        breaks_in_ha_version=breaks_in_ha_version,
+        is_fixable=True,
```

Comment:
Fixable means there is a config flow attached to it. That is not the case here, the user has to manually fix this.

Reply:
```suggestion
        is_fixable=False,
```